File: impl/php/fizzbuzz.php
<?php
declare(strict_types=1);

function word(int $n) {
    $is_fizz = $n % 3 == 0;
    $is_buzz = $n % 5 == 0;
    if ($is_fizz && $is_buzz) {
        return 'fizz buzz';
    } elseif ($is_fizz) {
        return 'fizz';
    } elseif ($is_buzz) {
        return 'buzz';
    } else {
        return (string)$n;
    }
}

function say(int $n) {
    echo word($n), "\n";
}

function play(int $n) {
    for ($i = 1; $i <= $n; $i++) {
        say($i);
    }
}

function read() {
    return intval(stream_get_contents(STDIN));
}

play(read());

File: impl/c/fizzbuzz.c
#include <stdio.h>

void say(int n) {
	bool is_fizz = n % 3 == 0;
	bool is_buzz = n % 5 == 0;
	if (is_fizz && is_buzz) {
		puts("fizz buzz");
	} else if (is_fizz) {
		puts("fizz");
	} else if (is_buzz) {
		puts("buzz");
	} else {
		printf("%d\n", n);
	}
}

void play(int n) {
	for (int i = 1; i <= n; i++) {
		say(i);
	}
}

int main() {
	int n;
	scanf("%d", &n);
	play(n);
}
